server/PythonClient/Client.py

```diff
@@ -0,0 +1,339 @@
+
+        # parse segment JSON payload data
+        parsed_segments = json.loads(segments)
+
+        return RecievedPayload(payloadData, self.readSegments(reciever, parsed_segments))
+
+    def readSegments(self, reciever, parsed_segments):
+        segmentDict = {}
+        for segment in parsed_segments:
+            segmentKey = next(iter(segment.keys()))
+            segmentMeta = next(iter(segment.values()))
+            size = int(segmentMeta["size"])
+            file_name = ""
+            if Segment._META_KEY_FILE_NAME in segmentMeta:
+                file_name = segmentMeta[Segment._META_KEY_FILE_NAME]
+            else:
+                file_name = str(uuid.uuid1())
+
+            fileFullPath = TMP_DIR + file_name
+            f = open(TMP_DIR + file_name, 'wb')
+            buffer = 25
+            remaining = size
+            l = reciever(min(buffer, remaining))
+            while (l):
+                remaining -= len(l)
+                f.write(l)
+                l = reciever(min(buffer, remaining))
+            f.close()
+            segmentDict[segmentKey] = RecivedSegment(segmentMeta, fileFullPath)
+        return segmentDict
+
+
+class RecivedSegment:
+    def __init__(self, segmentMeta, fileName):
+        self.segmentMeta = segmentMeta
+        self.fileName = fileName
+
+
+class RecievedPayload:
+    def __init__(self, payloadData, segments):
+        self.payloadData = payloadData
+        self.segments = segments
+
+
+class Segment(metaclass=abc.ABCMeta):
+
+    _META_KEY_SIZE = "size"
+    _META_KEY_MIME_TYPE = "mime_type"
+    _META_KEY_FILE_NAME = "file_name"
+
+    def __init__(self):
+        super().__init__()
+        self.segmentMeta = {}
+
+    @ abc.abstractmethod
+    def writeToStream(self):
+        pass
+
+    def getSegmentSize(self):
+        return self.segmentMeta.get(self._META_KEY_SIZE)
+
+    def getMeta(self):
+        return self.segmentMeta
+
+    def setSegmentSize(self, size: Type[int]):
+        self.segmentMeta[self._META_KEY_SIZE] = size
+
+    def setSegmentMimeType(self, mimeType: Type[str]):
+        self.segmentMeta[self._META_KEY_MIME_TYPE] = mimeType
+
+    def setSegmentFilename(self, fileName: Type[str]):
+        self.segmentMeta[self._META_KEY_FILE_NAME] = fileName
+
+
+class JsonSegment(Segment):
+
+    def __init__(self, jsonString: str):
+        super().__init__()
+        self.jsonString = jsonString
+        self.setSegmentSize(len(jsonString))
+        self.setSegmentMimeType("application/json")
+
+    def writeToStream(self, sendFunction: SendFunction):
+        sendFunction(bytes(self.jsonString, ENCODING))
+
+
+class FileSegment(Segment):
+
+    def __init__(self, file: BufferedReader):
+        super().__init__()
+        self._file = file
+        self.setSegmentSize(os.path.getsize(file.name))
+        self.setSegmentMimeType(mimetypes.guess_type(file.name)[0])
+        fileName = file.name.split("/")
+        self.setSegmentFilename(fileName[len(fileName) - 1])
```

Comment:
FYI i python kan du:
```suggestion
        self.setSegmentFilename(fileName[-1])
```